EventFilter/SiStripRawToDigi/interface/SiStripFEDBuffer.h

```diff
@@ -270,13 +271,19 @@ namespace sistrip {
 
   inline bool FEDBSChannelUnpacker::hasData() const
     {
-      return (currentWordOffset_<channelPayloadOffset_+channelPayloadLength_);
+      const uint16_t nextChanWordOffset = channelPayloadOffset_+channelPayloadLength_;
+      if ( currentWordOffset_ + 1 < nextChanWordOffset ) {
+        return true; // fast case: 2 bytes always fit an ADC (even if offset)
+      } else { // close to end
+        const uint16_t plusOneBitOffset = currentLocalBitOffset_+bitOffsetIncrement_;
+        const uint16_t plusOneWordOffset = currentWordOffset_ + plusOneBitOffset/8;
```

Comment:
This magic number "8" appears rather often in this class.
It could be more conveniently defined somewhere, perhaps as a ```sizeof()``` a given data word

Reply:
In this context it is the number of bits in a byte of raw data (`uint8_t`), so `sizeof` will not work (it gives the size of a type in bytes). I could define a constant, e.g. `constexpr uint16_t BITS_PER_BYTE = 8;`, but it would be only for readability.

Reply:
That would help, indeed